backend/crud.py

```diff
@@ -0,0 +1,211 @@
+from typing import Dict
+from sqlalchemy.orm import Session
+from sqlalchemy import select, update
+
+from .models import (
+    AttrType,
+    Attribute,
+    AttributeDefinition,
+    BoundFK,
+    Entity,
+    Schema,
+    Value,
+    val_map,
+    type_map
+)
+
+from .schemas import (
+    SchemaCreateSchema,
+    SchemaUpdateSchema,
+    AttributeCreateSchema
+)
+
+
+def create_attribute(db: Session, data: AttributeCreateSchema) -> Attribute:
+    exists = db.execute(select(Attribute).where(Attribute.name == data.name)).scalar()
+    if exists:
+        # TODO custom exceptions
+        raise Exception(f'Attribute with name `{data.name}` already exists')
+    a = Attribute(name=data.name, type=data.type)
+    db.add(a)
+    db.commit()
+    return a
+
+
+def create_schema(db: Session, data: SchemaCreateSchema) -> Schema:
+    # TODO require lock here and in update?
```

Comment:
It's probably unlikely that there are simultaneous requests for creating schemas, but it's better to be safe than sorry. I would use a lock here :smiley:

Reply:
My reason to put lock here (and in `update_schema()`) was that there might be simultaneous requests for creating and updating schemas which may lead to having multiple schemas with same name/slugs